Source/ACE/Network/GameAction/Actions/GameActionAutonomousPosition.cs

```diff
@@ -14,6 +16,17 @@ public static void Handle(ClientMessage message, Session session)
             var forcePositionTimestamp = message.Payload.ReadUInt16();
             message.Payload.ReadByte();
             session.Player.UpdatePosition(position);
+
+            if (session.Player.Statemachine.CurrentState != (int)MovementStates.Moving) return;
+            if ((Math.Abs(session.Player.PhysicsData.Position.SquaredDistanceTo(session.Player.MoveToPosition)) <= session.Player.ArrivedRadiusSquared))
+            {
+                session.Player.Statemachine.ChangeState((int)MovementStates.Arrived);
```

Comment:
I don't like hitting the StateMachine of the Player from here.  Perhaps we have a session.Player.OnArrived() method?

We'd also have a session.Player.IsPerformingMoveAction that his the state machine.

edit: even better: session.Player.OnAutonomousMove() that does it all.  It would also do the call to UpdatePosition, I think.